applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClient.java

```diff
@@ -75,60 +79,87 @@ public static Builder builder() {
+     * If {@link ConfigurationSetting#etag() etag} is specified, the configuration value is updated if the current setting's
      * etag matches. If the etag's value is equal to {@link ConfigurationClient#ETAG_ANY}, the setting will always be
      * updated.
      *
-     * <p>
-     * The label value for the ConfigurationSetting is optional.
-     *
      * @param setting The configuration setting to create or update.
      * @return ConfigurationSetting that was created or updated.
      * @throws NullPointerException If {@code setting} is {@code null}.
-     * @throws IllegalArgumentException If {@link ConfigurationSetting#key()} is {@code null} or an empty string.
-     * @throws ServiceRequestException If the {@link ConfigurationSetting#etag()} was specified, is not
-     * {@link ConfigurationClient#ETAG_ANY}, and the current configuration
-     * value's etag does not match.
+     * @throws IllegalArgumentException If {@link ConfigurationSetting#key() key} or {@link ConfigurationSetting#value() value}
+     * are {@code null} or an empty string.
+     * @throws ServiceRequestException If the {@link ConfigurationSetting#etag() etag} was specified, is not
+     * {@link ConfigurationClient#ETAG_ANY}, and the current configuration value's etag does not match.
      */
     public Mono<Response<ConfigurationSetting>> setSetting(ConfigurationSetting setting) {
-        ConfigurationSetting result = validateSetting(setting);
+        ConfigurationSetting result = validateSetting(setting, true);
 
         return service.setKey(serviceEndpoint, result.key(), result.label(), result, getETagValue(result.etag()), null);
     }
 
+    /**
+     * Updates an existing configuration value in the service with the given key. The setting must already exist.
+     *
+     * <p>
+     * Partial updates are not supported.
+     *
+     * @param key The key for the configuration setting to update.
+     * @param value The updated value of this configuration setting.
+     * @return ConfigurationSetting that was updated.
+     * @throws IllegalArgumentException If {@code key} or {@code value} are {@code null} or an empty string.
+     * @throws ServiceRequestException If a ConfigurationSetting with the key does not exist or the configuration value
+     * is locked.
+     */
+    public Mono<Response<ConfigurationSetting>> updateSetting(String key, String value) {
+        return updateSetting(new ConfigurationSetting().key(key).value(value));
+    }
+
     /**
      * Updates an existing configuration value in the service. The setting must already exist.
      *
      * <p>
-     * The label value for the ConfigurationSetting is optional.
+     * Partial updates are not supported.
      *
      * <p>
-     * If {@link ConfigurationSetting#etag()} is specified, the configuration value is only updated if it matches.
+     * If {@link ConfigurationSetting#etag() etag} is specified, the configuration value is only updated if it matches.
```

Comment:
You're on a paragraph frenzy. I recommend generating the HTML output and ensuring you like what you see. I suspect it'll look a little sparse.

Reply:
Ahaha. I got excited with those `<p>` tags. I am too used to lists and short statements. 🤣Oops.